### PR TITLE
[DR-3022] TDR uses B2C auth in developer, test deployments

### PR DESCRIPTION
**Pull request by okotsopoulos**
https://broadworkbench.atlassian.net/browse/DR-3022

For consistency among TDR deployments, we wanted to bring our developer and test deployments up-to-date and use B2C for authentication rather than Google OAuth2.  This discrepancy recently resulted in a new UI feature merged that worked on dev deployments with Google OAuth2, but did not work in the other environments with B2C Auth.

We've also been running into issues with Google OAuth2 on our personal deployments lately ([Slack](https://broadinstitute.slack.com/archives/C01VBGH431S/p1682708316066789?thread_ts=1682694459.993999&cid=C01VBGH431S)).  Rather than work to fix, we're taking the opportunity to make authentication consistent.

**1. Initial clean up of unneeded deployments**

To minimize the number of configurations that we need to keep up to date, I removed configurations for past team members, most of whom no longer had active deployments.  I also removed the `az` (Azure) configuration: @nmalfroy says "For azure features, those just live on dev and prod now".

If this clean-up is approved, I will then want to clean up the deployed resources for `az`, `cl` (Christine Lu), and `dd` (Dan Deitterich) namespaces. **ETA** -- done.

**2. Switching remaining deployments to B2C**

For the deployments that remained -- current team members, integration tests, and perf tests -- I switched over their OIDC configuration to use B2C authentication, which included the removal of `existingOauthClientSecretSecretKey`.  I found that if you leave this in, logging into your TDR UI will fail with a `400 Bad Request` error: `Public clients should not send a client_secret when redeeming a publicly acquired grant.`

I applied these changes to my dev environment.  Log-in and other interactions all work with B2C auth: https://jade-ok.datarepo-dev.broadinstitute.org/

Note: others will need to apply their own changes for them to take effect, but by checking them in I wanted to make sure that if a new member joined and copied another's directory as a starting place, they'd be getting B2C auth by default.

**Follow-on work**

- As mentioned above, cleaning up deployed resources associated with removed configs
- In `jade-data-repo` updating onboarding documentation and needed configs to default to B2C auth locally

### CODE REVIEW

**Review by nmalfroy — APPROVED** [review]
thanks for this!  note: we'll still need to apply to our own envs but it's great it'll be from a reasonable place!

**Review by samanehsan — APPROVED** [review]

**Comment by okotsopoulos** [comment]
Note for posterity:

I documented my experience trying to clean up `az`, `cl`, and `dd` old deployments in Slack - https://broadinstitute.slack.com/archives/C01VBGH431S/p1683058657724509?thread_ts=1683044297.763199&cid=C01VBGH431S

TL;DR -- `helmfile delete` was insufficient for all namespaces, but I was able to go to the GCP console and clean up the remaining releases there.  I have not yet touched [dev-specific cloud DBs](https://console.cloud.google.com/sql/instances/jade-postgres-11-8a00fd4d3b/databases?authuser=1&project=broad-jade-dev) but those could probably stand to be cleaned up too.
